### PR TITLE
CMake Support changes

### PR DESCRIPTION
**Pull request by agaida**
Bumped cmake_minimum_required 3.0.2
Added CMAKE_BUILD_TYPE, if not set
Added check for compiler support
Added set CXX flags

@palinek @luis-pereira  what do you think?

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
+CHECK_CXX_COMPILER_FLAG("-std=c++0x" COMPILER_SUPPORTS_CXX0X)
+if(COMPILER_SUPPORTS_CXX11)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+elseif(COMPILER_SUPPORTS_CXX0X)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x")
+else()
+    message(FATAL "The compiler ${CMAKE_CXX_COMPILER} has no C++11 support. C++11 support is required")
+endif()
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
```
Comment (f2404) [inline CMakeLists.txt]: Why is this needed? `-std` has already been set.

Comment (agaida) [inline CMakeLists.txt]: Bah - an insert to much, thanks

**Comment by luis-pereira** [comment]
@agaida With that line removal, it's fine. Can you do the same in qtermwidget ?

**Comment by agaida** [comment]
@luis-pereira sure, it is prepared

**Comment by agaida** [comment]
not my day - to dumb for git ...

**Comment by luis-pereira** [comment]
@agaida moving patches between repos, it's hard ;)
